source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -373,6 +373,26 @@ void AuthenticatorImpl::doneWithStatus(const Status& status) {
   ENVOY_LOG(debug, "{}: JWT token verification completed with: {}", name(),
             ::google::jwt_verify::getStatusString(status));
 
+  if (Status::Ok != status) {
+    // Forward the failed status to dynamic metadata
+    ENVOY_LOG(info, "status is: {}", ::google::jwt_verify::getStatusString(status));
+
+    if (!jwks_cache_.findByProvider(*provider_)
+             ->getJwtProvider()
+             .failed_status_in_metadata()
+             .empty()) {
+
+      ProtobufWkt::Struct failed_status;
+      auto& failed_status_fields = *failed_status.mutable_fields();
```

Comment:
why don't we log both code and status string as

```
"code":  enumToInt(status)
"message": ::google::jwt_verify::getStatusString(status)
```

If the code using enum code status is not compiled with the same jwt_verify_lib version as envoy,  you may have problem if you just use enum as  jwt_verify_lib may change enum code.